packages/go_router/lib/src/state.dart

```diff
@@ -74,6 +76,9 @@ class GoRouterState {
   /// ```
   final ValueKey<String> pageKey;
 
+  /// The current matched top route
```

Comment:
This needs more description. May be mention what this mean if you this is a this GoRouteState represent a shellroute.

Reply:
I've updated the documentation with the following, is this what you were thinking of?

```dart
  /// The current matched top route associated with this state.
  ///
  /// If this state represents a [ShellRoute], the top [GoRoute] will be the current
  /// matched location associated with the [ShellRoute]. This allows the [ShellRoute]'s
  /// associated [GoRouterState] to be uniquely identified using [GoRoute.name]
  final GoRoute? topRoute;
```